### PR TITLE
[#1358] build Request ZEC flow

### LINKED ISSUE #1358: build Request ZEC flow
- as a user I can generate a payment QR code with my address, memo copy and an amount 
- I can share the QR code (with the payment URI encoded) with anyone and when they scan it, their Zashi wallet generates a transaction proposal with the data encoded in the QR code

**Access Point from the Receive screen:**

- when a user taps on the Request button on the Receive screen, we should route them to the Request ZEC flow (depending on whether they tapped on a shielded address or a transparent address, we should create a payment URI for the corresponding address)
- shielded: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=1956-5188&t=fVE9gwu9KMMADuX0-4
- transparent: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=1956-5412&t=fVE9gwu9KMMADuX0-4

**Request ZEC flow:**

**Step 1:** Amount screen - a user can set an amount for the request and flip the currencies by tapping on the 🔄 button (https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3077-11494&t=fVE9gwu9KMMADuX0-4)

**Step 2:** Memo text - a user can add text to the memo field (only in case of a shielded request) (https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=1956-4368&t=fVE9gwu9KMMADuX0-4)

**Step 3:** after tapping on the Request button, a payment URI is encoded in a QR code, and a user can share the QR code with a message via the native sharing options:
- Request screen for Shielded: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3077-11392&t=fVE9gwu9KMMADuX0-4
- Request screen for Transparent: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3077-11443&t=fVE9gwu9KMMADuX0-4
- Sharing options: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=1956-5092&t=fVE9gwu9KMMADuX0-4

**3rd Party Scanner Used:** 

- In case a user scans the QR code with their camera app, we should launch Zashi, and display a warning screen to them: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3152-6450&t=fVE9gwu9KMMADuX0-4
- When a user taps on the Rescan in Zashi button, we should route them to the Scan screen

**Payer/Consumer Experience:** 

- once a user scans the QR code with the Zashi scan UI, we should route them to a transaction confirmation screen:
- Shielded tx, Address Collapsed, Contact Not Saved: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3173-8460&t=fVE9gwu9KMMADuX0-4
- Shielded tx, Address Expanded, Contact Not Saved: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3173-8301&t=fVE9gwu9KMMADuX0-4
- Transparent, Contact Not Saved: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3175-6237&t=fVE9gwu9KMMADuX0-4
- Shielded tx, Address Collapsed, Contact Saved: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3175-6029&t=fVE9gwu9KMMADuX0-4
- Shielded tx, Address Expanded, Contact Saved: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3175-6129&t=fVE9gwu9KMMADuX0-4
- Transparent tx, Contact Saved: https://www.figma.com/design/ItxXKIvUZRpdabGhMAlmPs/PR-Designs-(2024)?node-id=3175-6337&t=fVE9gwu9KMMADuX0-4

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #1358 

- zip321 payment uri is generated in a form or a QR code, can be shared or terminated
- ZecKeyboard fixes or reported issues
- REQUEST title for the whole screen fix
- Padding fixes in the request zec flow
- Consumer part of the request zec implemented
- The code is scanned and dedicated confirmation screen is shown
- When system camera is used, the blocking view is shown
- zcash: scheme activated for Zashi
- The code is finished for the request zec (both generate & scan) except localized texts (it'll be done in a separate PR)

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: modules/Package.swift
Comment (HonzaR) [inline modules/Package.swift:92]: Should we rather remove this line?

File: modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift
Comment (HonzaR) [inline modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift:53]: The same comment here. Should we remove this, or do we have a plan with the commented-out code?

**Review by HonzaR — APPROVED** [review]
I see several commented-out code fragments and TODOs without linking Github issues in this PR. But we don't want to block this PR, so it's up to you if those need to be addressed now or later when we'll go with these changes to the main branch. utACK.

File: modules/Package.swift
Comment (LukasKorba) [reply modules/Package.swift:92]: All commented code will be removed in a followup ticket (once other PRs are merged to the feature branch). In this specific case I need to share builds with Core team with the note commitment tree fix so it's going to be switched back to main SDK (2.2.4) with the fix.

File: modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift
Comment (LukasKorba) [reply modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift:53]: All commented code will be removed in a followup ticket (once other PRs are merged to the feature branch).

**Comment by LukasKorba** [comment]
> I see several commented-out code fragments and TODOs without linking Github issues in this PR. But we don't want to block this PR, so it's up to you if those need to be addressed now or later when we'll go with these changes to the main branch. utACK.

Thank you, due to nature of several features not being done or blocked, the commented code is just waiting on merge and unblocked state. This PR is to feature branch only so the main goal is to clean up the code for the final main PR.

**Pull request merged**
